sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -272,27 +282,33 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
     ctx match {
       case table: InsertIntoTableContext =>
         val (relation, cols, partition, ifPartitionNotExists) = visitInsertIntoTable(table)
-        InsertIntoStatement(
-          relation,
-          partition,
-          cols,
-          query,
-          overwrite = false,
-          ifPartitionNotExists)
+        withIdentClause(relation, ident => {
+          InsertIntoStatement(
```

Comment:
since we already have `UnresolvedRelationIdentifierClause`, we can use it for INSERT.

Reply:
We only have UnresolvedRelationIdentifierClause because I haven't figured MERGE out.